partiql-types/src/main/java/org/partiql/value/PartiQLData.java

```diff
@@ -0,0 +1,158 @@
+     */
+    public boolean isNullValue();
+
+    /**
+     * Determines whether the current value is the missing value. Similarly, one can invoke {@link #getType()}.
+     */
+    public boolean isMissingValue();
+
+    /**
+     * @return the type of the data at the cursor.
+     */
+    @NotNull
+    public PartiQLValueType getType();
+
+    /**
+     * @return the field name of the current value; or null if there is no valid current value or if the current value
+     * is not a field of a struct.
+     */
+    public String getFieldName();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#STRING},
+     * {@link PartiQLValueType#CHAR}, or {@link PartiQLValueType#SYMBOL}.
+     */
+    @NotNull
+    String getStringValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#BOOL}.
+     */
+    public boolean getBoolValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#BINARY},
+     * {@link PartiQLValueType#BLOB}, or {@link PartiQLValueType#CLOB}.
+     */
+    public byte[] getBytes();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#INT8} or
+     * {@link PartiQLValueType#BYTE}.
+     */
+    public byte getByteValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#DATE}.
+     */
+    @NotNull
+    public Date getDate();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#TIME}.
+     */
+    @NotNull
+    public Time getTime();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#TIMESTAMP}.
+     */
+    @NotNull
+    public Timestamp getTimestamp();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#INTERVAL}.
+     */
+    @Deprecated
+    public long getIntervalValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#INT16}.
+     */
+    public short getShortValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#INT32}.
+     */
+    public int getIntValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#INT64}.
+     */
+    public long getLongValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#INT}.
+     */
+    @NotNull
+    public BigInteger getBigIntValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#FLOAT32}.
+     */
+    public float getFloatValue();
+
+    /**
+     * This is only applicable when the current value's type is {@link PartiQLValueType#FLOAT64}.
+     */
+    public double getDoubleValue();
```

Comment:
Interesting seeing the Java type names appear in the method names. Let's discuss. I'd prefer the partiql type names to be the method names since this is the interface (and may appear in other language implementations). 

I advise to say `public long getInt64Value()` or something similar just using the PartiQLValue names exactly